docs/userguide/main.md

```diff
@@ -342,3 +342,28 @@ spec:
 ```
 
 To perform a rolling upgrade on the cluster, simply change this version and the Operator will perform a rolling upgrade. Downgrades and upgrades that span more than one major version are not supported, as this will put the OpenSearch cluster in an unsupported state. If you are using emptyDir storage for data nodes, it is recommended to set `general.drainDataNodes` to `true`, otherwise you might lose data.
+
+## Set Java heap size
+
+To configure the amount of memory allocated to the OpenSearch nodes, configure the heap size using the JVM args. This operation is expected to have no downtime and the cluster should be operational.
+
+Recommendation: Set to half of memory request
+
+```yaml
+spec:
+    nodePools:
+      - component: nodes
+        replicas: 3
+        diskSize: "10Gi"
+        JVM: -Xmx1024M -Xms1024M
```

Comment:
IMHO you should use the RAM percentage parameters which have the advantage that you don't need to update them every time you change the `resources` below: https://www.baeldung.com/java-jvm-parameters-rampercentage

and you could actually just set them by default (e.g. in the docker image or on the `StatefulSet` of opensearch) so that users of the operator don't need to deal with this.

Reply:
Agreed. But the point of this PR its just to update the userguide, because that specific information was missing. 

Please add an example to the userguide using RAM percentage parameters aswell if you can. I am not familiar with it.

Reply:
@rursprung Good idea with using the percentage parameter. As you offered, could you please open a new issue for this?

Reply:
done: #171